src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/TaskControllerTest.java

```diff
@@ -145,20 +153,57 @@ public void should_return_503_for_work_in_progress_endpoints() {
             .body("error", equalTo(HttpStatus.SERVICE_UNAVAILABLE.getReasonPhrase()))
             .body("status", equalTo(HttpStatus.SERVICE_UNAVAILABLE.value()))
             .body("message", equalTo(responseMessage));
+    }
 
-        result = given()
+    @Test
+    public void should_return_a_204_when_completing_and_204_when_completing_an_already_completed_task() {
+        String ccdId = ccdIdGenerator.generate();
+
+        List<CamundaTask> tasks = given
+            .iCreateATaskWithCcdId(ccdId)
+            .and()
+            .iRetrieveATaskWithProcessVariableFilter("ccdId", ccdId);
+
+        if (tasks.size() > 1) {
+            fail("Search was not an exact match and returned more than one task:" + "used:" + ccdId);
+        }
+
+        String taskId = tasks.get(0).getId();
+
+        Response result = given()
             .contentType(MediaType.APPLICATION_JSON_VALUE)
             .headers(authorizationHeadersProvider.getAuthorizationHeaders())
             .when()
-            .post("/task/{task-id}/complete", taskId);
+            .post("task/{task-id}/complete", taskId);
 
         result.then().assertThat()
-            .statusCode(HttpStatus.SERVICE_UNAVAILABLE.value())
+            .statusCode(HttpStatus.NO_CONTENT.value());
+
+        List<HistoryVariableInstance> historyVariableInstances = given()
+            .contentType(APPLICATION_JSON_VALUE)
+            .baseUri(camundaUrl)
+            .when()
+            .get("/history/variable-instance?taskIdIn=" + taskId)
+            .prettyPeek()
+            .then()
+            .statusCode(HttpStatus.OK.value())
             .and()
+            .extract()
+            .jsonPath().getList("", HistoryVariableInstance.class);
+
+        List<HistoryVariableInstance> taskState = historyVariableInstances.stream()
+            .filter(historyVariableInstance -> historyVariableInstance.getName().equals("taskState"))
+            .collect(Collectors.toList());
```

Comment:
this seems to work but perhaps we want to retrieve the task again using GET /task/{task-id}
and assert on the taskState however to do this it will need to wait until RWA-50 is merged as it contains the task mapping and stitching of the localvariables

Reply:
Think we talk about why I would rather do this this way.